### ISSUE TITLE
CT-scan preprocessing

### ISSUE THREAD

**Issue by kiwibruyant** [voxelmorph/voxelmorph#405]
Hello,

I'm working on registration on longitudinal CT-Scan of chest/abdomen/lungs and I'm wondering if I should do the same preprocessing steps used in the voxel morph paper for the brain; i.e, affine registration and especially skull stripping (in this case it would be rib cage stripping) ?

Thanks in advance,

**Comment by adalca** [comment]
no, those are not necessary -- you just have to choose the right losses for your data. I'd recommending rescaling your data to be roughly in the (0,1) or (-1,1) range and starting from there.

**Comment by kiwibruyant** [comment]
@adalca Thank you very much. I have another questions. To evaluate voxelmorph's performances with DICE score you are using segmentations labels of anatomical regions/area, I'm wondering how do you obtain them on the warped images obtained after the registration process, are you doing it manually (by an expert for instance) or via an automatic algorithm ?

**Comment by adalca** [comment]
are you asking how to segment the original images or how to warp segmentations?

to segment original images we sometimes have manual segmentations or something done with algorithms (see e.g. https://surfer.nmr.mgh.harvard.edu/fswiki/SynthSeg)

to propagate (warp) label maps you can take a look at how its done in the test.py script, basically we just use the deformation field to warp the label maps.

Note that Dice is a person's name (Lee R. Dice), not an acronym (so not DICE)

**Comment by pangyuteng** [comment]
Extremely cool project and thanks for sharing the code and weights to the public. Thank you @adalca !!
Kudos to @ahoopes for adding the `rescale` flag in class `Transform` (8f3f519) ! 🙏🙏 

--
for those interested in assessing voxelmorph with CT images, am providing my "spaghetti" attempt [here](https://github.com/pangyuteng/hello-voxelmorph/blob/78e21af74badfbb5d9d76eff49ba1b323df92aaf/voxelmorph/scripts/register.py).  With the `rescale` flag set, you can then transform using a higher resolution moving image. while computing the registration field using down-sampled fixed and moving pairs. (no more OOM! or waiting for 2+hrs using conventional registration techniques PER PAIR 🤯 )

Currently still exploring the code, and already am stunned by the results (n=1) using the shape-variant SynthMorph weights. 
 Below shows the chest CTs under total-lung-capacity (TLC, as moving, left), residual-volume (RV, as fixed, middle), and TLC transformed to the RV space (moved, right).  We zoom in to the lower part of the right lung, given that is where the major displacement should occur from TLC to RV.  In the sagittal views, right minor fissure (red arrows and cross hair) are somewhat visually aligned and you can see the pulmonary vessels somewhat have the similar pattern, indicating good alignment.

![image](https://github.com/voxelmorph/voxelmorph/assets/5231403/bd2c4174-d5f1-49e9-b5d8-ebae4d8b7768)
